dm/syncer/syncer.go

```diff
@@ -931,10 +940,15 @@ func (s *Syncer) addJob(job *job) error {
 	case flush:
 		s.addCount(false, adminQueueName, job.tp, 1, job.targetTable)
 		s.jobWg.Add(1)
+		job.wg.Add(s.cfg.WorkerCount)
 		s.dmlJobCh <- job
-		s.jobWg.Wait()
-		s.addCount(true, adminQueueName, job.tp, 1, job.targetTable)
-		return s.flushCheckPoints()
+		var err error
+		if job.sync {
+			err = s.flushCheckPoints()
```

Comment:
If my review is correct, job.sync means we should wait for all jobs flushed, but here s.flushCheckpoints will call `doFlushCheckPointsAsync` with empty waitgroup, how do we ensure that all jobs are flushed?

Reply:
+1

Reply:
by `s.jobWg.Wait()` in line 949?

Reply:
for `flushCheckPoints` function, it will wait until errCh return, and this errCh will emit data out when `w.cp.FlushPointsExcept` return.
```
func (s *Syncer) flushCheckPoints() error {
	errCh := make(chan error, 1)
	// use math.MaxInt64 to clear all the data in causality component.
	s.doFlushCheckPointsAsync(nil, errCh, math.MaxInt64)
	return <-errCh
}
```